keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/MatrixDeterminantVertexTest.java

```diff
@@ -0,0 +1,111 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary;
+
+import static io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.TensorTestOperations.finiteDifferenceMatchesReverseModeGradient;
+
+import org.apache.commons.math3.exception.MathArithmeticException;
+import org.apache.commons.math3.linear.SingularMatrixException;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import io.improbable.keanu.algorithms.variational.optimizer.Optimizer;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.TensorMatchers;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.Differentiator;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+
+public class MatrixDeterminantVertexTest {
+    @Test
+    public void calculatesDeterminant() {
+        final DoubleVertex input = new ConstantDoubleVertex(DoubleTensor.create(new double[]{1, 2, 3, 4}, 2, 2));
+        Assert.assertThat(input.matrixDeterminant().getValue(), TensorMatchers.isScalarWithValue(-2d));
+    }
+
+    @Test
+    public void calculatesDeterminantOnLargeMatrix() {
+        final double[] values = new double[]{
+            1, 8, 4, 5, 6,
+            1, 100, 232, 4, 15,
+            -10, 3, 57, 68, 10,
+            7, 8, 9, 20, 10,
+            12, 32, 43, -2, 5};
+        final DoubleVertex input = new ConstantDoubleVertex(DoubleTensor.create(values, 5, 5));
+        Assert.assertThat(input.matrixDeterminant().getValue(), TensorMatchers.isScalarWithValue(Matchers.closeTo(6120880d, 1e-5)));
```

Comment:
Determinant calculations end up with big numbers for larger matrices - this calculation in particular ended up being not exact (so an exact match would fail)